app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  before_action :authorize_user
+
+  def show
```

Comment:
hmm. We can still use the `username` field as the `id` parameter. This would be expected behavior. 

here's how to tackle this:

```
class User < ApplicationRecord
  def to_param
    self.username
  end
```

With this in place when you say `user_path(@user)`, Rails route helpers will place the `username` in as the `id` param.

Once that's done, this controller can be updated to do a find based on what is in `params[:id]`

Reply:
@dpickett This new `to_param` method isn't acting as we expect it to.

When we make the update, `user_path(@user)` throws an error saying `No route matches {:action=>"show", :controller=>"users", :id=>nil}, missing required keys: [:id]`. It also causes some cascading issues with the other code we had in our controller.